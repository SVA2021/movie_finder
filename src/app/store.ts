import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
