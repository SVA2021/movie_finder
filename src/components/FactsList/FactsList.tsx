import {Box, Typography, useTheme} from '@mui/material';
import {FC, memo, useEffect, useRef} from 'react';
import 'swiper/css';
import {Swiper, SwiperSlide} from 'swiper/react';
import {useAppDispatch, useAppSelector} from '../../app/hooks';
import {getMovieFactsAsync, selectDetails} from '../../features/movieFinder/movieFinderSlice';
import {TMovieExtraProps} from '../../features/movieFinder/movieFinderTypes';

export const FactsList: FC<TMovieExtraProps> = memo(({id}) => {

    const dispatch = useAppDispatch();
    const DetailsPage = useAppSelector(selectDetails);
    const MovieFacts = DetailsPage.facts?.items ?? [];
    const theme = useTheme();
    const swiperRef = useRef<any>();

    useEffect(() => {
        dispatch(getMovieFactsAsync(id))
    }, [id]);

    return (
        <Box>
            {MovieFacts.length > 0
                ? <Swiper
                    onSwiper={(swiper) => swiperRef.current = swiper}
                    slidesPerView={1}
                    spaceBetween={8}
                    breakpoints={
                        {
                            [theme.breakpoints.values.md]: {
                                slidesPerView: 3,
                                spaceBetween: 16,
                            },
                            [theme.breakpoints.values.lg]: {
                                slidesPerView: 3,
                                spaceBetween: 24,
                            },
                        }
                    }
                >
                    {
                        MovieFacts.map((item, index) =>
                            <SwiperSlide key={index} >
                                <Box sx={{cursor: 'pointer'}} >
                                    <Typography
                                        pb={2}
                                        variant="h4"
                                        textTransform={'capitalize'}
                                        component={'p'} color={'text.secondary'}
                                    >
                                        {item.type === 'FACT' ? 'факт' : 'слух'}
                                    </Typography>
                                    <Typography variant="body1" component={'p'} sx={{wordBreak: 'break-word', }}>
                                        {`${item.text}`}
                                    </Typography>
                                </Box>
                            </SwiperSlide>
                        )
                    }
                </Swiper>
                :
                <Typography variant="h3" component={'p'} fontStyle={'italic'}>
                    Нет данных
                </Typography>
            }
        </Box>
    );
});