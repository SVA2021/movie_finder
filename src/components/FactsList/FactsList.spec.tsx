import {render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {FactsList} from "./FactsList";

describe('render FactsList', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <FactsList />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('facts-list')).toMatchSnapshot();
  });

  it('check empty text', () => {
    setUp();
    expect(screen.getByText(/нет данных/i)).toBeInTheDocument();
  });

})