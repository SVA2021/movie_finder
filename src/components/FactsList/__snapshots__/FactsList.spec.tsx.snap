// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FactsList snapshot test 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="facts-list"
>
  <p
    class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
    data-testid="empty-data-box"
  >
    Нет данных
  </p>
</div>
`;
